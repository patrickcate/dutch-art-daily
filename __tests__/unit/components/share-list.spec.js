import ShareList from '@components/share-list.vue'

describe('ShareList Component', () => {
  let wrapper
  let options

  beforeEach(() => {
    options = {
      propsData: {
        direction: 'horizontal',
      },
    }

    wrapper = createWrapper(ShareList, options)
  })

  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('does not render correctly', () => {
    const wrapper = createWrapper(
      ShareList,
      {
        ...options,
      },
      {
        getters: {
          getArtworkById: () => () => {
            return null
          },
        },
      }
    )

    expect(wrapper).toMatchSnapshot()
  })
})
