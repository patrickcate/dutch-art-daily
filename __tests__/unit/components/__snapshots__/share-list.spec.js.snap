// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareList Component does not render correctly 1`] = ``;

exports[`ShareList Component renders correctly 1`] = `
<ul class="share-list share-list--horizontal">
  <li class="share-list__item share-list__item--facebook"><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A//www.dutchartdaily.com/01-01" class="share-list__item-link"><span class="u-sr-only">Share on Facebook</span> <svg width="1em" height="1em" class="svg-icon share-list__icon" role="presentation">
        <use></use>
      </svg></a></li>
  <li class="share-list__item share-list__item--twitter"><a href="https://twitter.com/intent/tweet?url=https%3A//www.dutchartdaily.com/01-01&amp;text=The%20Milkmaid%20by%20Johannes%20Vermeer" class="share-list__item-link"><span class="u-sr-only">Share on Twitter</span> <svg width="1em" height="1em" class="svg-icon share-list__icon" role="presentation">
        <use></use>
      </svg></a></li>
  <li class="share-list__item share-list__item--pinterest"><a href="https://pinterest.com/pin/create/link/?url=https%3A//www.dutchartdaily.com/01-01&amp;media=https%3A//www.dutchartdaily.com/photos/01-01/01-01--xs3-d0wAmzwM620.jpg&amp;description=The%20Milkmaid%20by%20Johannes%20Vermeer" class="share-list__item-link"><span class="u-sr-only">Share on Pinterest</span> <svg width="1em" height="1em" class="svg-icon share-list__icon" role="presentation">
        <use></use>
      </svg></a></li>
  <li class="share-list__item share-list__item--github"><a href="https://github.com/patrickcate/dutch-art-daily" class="share-list__item-link"><span class="u-sr-only">Star on GitHub</span> <svg width="1em" height="1em" class="svg-icon share-list__icon" role="presentation">
        <use></use>
      </svg></a></li>
</ul>
`;
